### ISSUE TITLE
mongodb input plugin not reporting replica-set data

### ISSUE THREAD

**Issue by Jaff** [influxdata/telegraf#2890]
### Relevant telegraf.conf:
abbreviated telegraf.conf, plus monitor_mongodb.conf, see https://gist.github.com/Jaff/3a2024afaf44b9b5a448090f46a94fc8

### System info:

[Include Telegraf version, operating system name, and other relevant details]
1.3.0, 1.3.1 on Ubuntu 14.04.4 (x86_64)
scalyr-agent v2.0.25 
### Steps to reproduce:

1. See related gist configs, and running 1.3.0 or 1.3.1 telegraf
2. ...

### Expected behavior:
Expect to collect detail about Replica-Set Status (per source code at https://github.com/influxdata/telegraf/blob/master/plugins/inputs/mongodb/mongostat.go#L98-L111)
### Actual behavior:
No evidence of this in scalyr2 /var/log/scalyr-agent-2/graphite_monitor.log
### Additional info:

[Include gist of relevant config, logs, etc.]

## Feature Request

Opening a feature request kicks off a discussion.

**Comment by Jaff** [comment]
Switched output from graphite to json, here is detail from mongodb plugin:
{"fields":{"active_reads":0,"active_writes":0,"commands_per_sec":228,"deletes_per_sec":0,"flushes_per_sec":0,"getmores_per_sec":147,"inserts_per_sec":1,"jumbo_chunks":0,"member_status":"PRI","net_in_bytes":1708761,"net_out_bytes":121581184,"open_connections":113,"percent_cache_dirty":1.5,"percent_cache_used":80.5,"queries_per_sec":5846,"queued_reads":1,"queued_writes":0,"repl_commands_per_sec":0,"repl_deletes_per_sec":0,"repl_getmores_per_sec":0,"repl_inserts_per_sec":0,"repl_lag":0,"repl_queries_per_sec":0,"repl_updates_per_sec":0,"resident_megabytes":17776,"state":"PRIMARY","ttl_deletes_per_sec":0,"ttl_passes_per_sec":0,"updates_per_sec":247,"vsize_megabytes":24083},"name":"mongodb","tags":{"hostname":"127.0.0.1:27017"},"timestamp":1496951100}
{"fields":{"avg_obj_size":343.33931732108806,"collections":299,"data_size":651408604055,"index_size":68940144640,"indexes":991,"num_extents":0,"objects":1897273546,"ok":1,"storage_size":198537789440,"type":"db_stat"},"name":"mongodb_db_stats","tags":{"db_name":"<redacted>","hostname":"127.0.0.1:27017"},"timestamp":1496951100}
{"fields":{"avg_obj_size":381.7231716593714,"collections":6,"data_size":24659707392,"index_size":102400,"indexes":5,"num_extents":0,"objects":64601023,"ok":1,"storage_size":25229033472,"type":"db_stat"},"name":"mongodb_db_stats","tags":{"db_name":"local","hostname":"127.0.0.1:27017"},"timestamp":1496951100}
So missing from graphite_monitor.log is member_status and state details.

**Comment by danielnelson** [comment]
Thanks for looking into this, can you add the graphite output too?  Easiest way to do this is with the file output configured with your graphite settings:
```toml
[[outputs.file]]
  files = ["stdout"]
  data_format = "graphite"
  prefix = "telegraf"
  template = "host.tags.measurement.field"
```

**Comment by arohter** [comment]
I believe graphite metric *values* are numeric only, so strings like "PRIMARY" won't make it in 😢

**Comment by Jaff** [comment]
Yes, we observed this in https://github.com/influxdata/influxdb/issues/1956 so it appears we are blocked. Can the graphite issue be re-opened as a feature request?

**Comment by Jaff** [comment]
Here's some output from file when configured as @danielnelson mentioned above:
telegraf.127_0_0_1:27017.mongodb.net_in_bytes 218 1496952960
telegraf.127_0_0_1:27017.mongodb.deletes_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.flushes_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.ttl_passes_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.percent_cache_dirty 0.9 1496952960
telegraf.127_0_0_1:27017.mongodb.commands_per_sec 2 1496952960
telegraf.127_0_0_1:27017.mongodb.queries_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.repl_commands_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.state "SECONDARY" 1496952960
telegraf.127_0_0_1:27017.mongodb.repl_getmores_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.repl_queries_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.vsize_megabytes 27070 1496952960
telegraf.127_0_0_1:27017.mongodb.jumbo_chunks 0 1496952960
telegraf.127_0_0_1:27017.mongodb.repl_updates_per_sec 149 1496952960
telegraf.127_0_0_1:27017.mongodb.member_status "SEC" 1496952960
telegraf.127_0_0_1:27017.mongodb.inserts_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.updates_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.repl_deletes_per_sec 59 1496952960
telegraf.127_0_0_1:27017.mongodb.open_connections 15 1496952960
telegraf.127_0_0_1:27017.mongodb.getmores_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.queued_reads 0 1496952960
telegraf.127_0_0_1:27017.mongodb.ttl_deletes_per_sec 0 1496952960
telegraf.127_0_0_1:27017.mongodb.repl_inserts_per_sec 31 1496952960
telegraf.127_0_0_1:27017.mongodb.resident_megabytes 21756 1496952960
telegraf.127_0_0_1:27017.mongodb.net_out_bytes 1709 1496952960
telegraf.127_0_0_1:27017.mongodb.percent_cache_used 89.5 1496952960
telegraf.127_0_0_1:27017.mongodb.active_reads 1 1496952960
telegraf.127_0_0_1:27017.mongodb.repl_lag 0 1496952960
telegraf.127_0_0_1:27017.mongodb.active_writes 0 1496952960
telegraf.127_0_0_1:27017.mongodb.queued_writes 0 1496952960

**Comment by danielnelson** [comment]
Thanks @arohter, appreciate the help.
